types/hapi__joi/index.d.ts

```diff
@@ -152,1245 +226,1757 @@ export interface DomainOptions {
+         * String comparisons are by default case insensitive,
+         * see `boolean.sensitive()` to change this behavior.
+         * @param values - strings, numbers or arrays of them
+         */
+        falsy(...values: Array<string | number>): this;
+
+        /**
+         * Allows the values provided to truthy and falsy as well as the "true" and "false" default conversion
+         * (when not in `strict()` mode) to be matched in a case insensitive manner.
+         */
+        sensitive(enabled?: boolean): this;
+
+        /**
+         * Allows for additional values to be considered valid booleans by converting them to true during validation.
+         * String comparisons are by default case insensitive, see `boolean.sensitive()` to change this behavior.
+         * @param values - strings, numbers or arrays of them
+         */
+        truthy(...values: Array<string | number>): this;
+    }
+
+    interface NumberSchema extends AnySchema {
+        /**
+         * Specifies that the value must be greater than limit.
+         * It can also be a reference to another field.
+         */
+        greater(limit: number | Reference): this;
+
+        /**
+         * Requires the number to be an integer (no floating point).
+         */
+        integer(): this;
+
+        /**
+         * Specifies that the value must be less than limit.
+         * It can also be a reference to another field.
+         */
+        less(limit: number | Reference): this;
+
+        /**
+         * Specifies the maximum value.
+         * It can also be a reference to another field.
+         */
+        max(limit: number | Reference): this;
+
+        /**
+         * Specifies the minimum value.
+         * It can also be a reference to another field.
+         */
+        min(limit: number | Reference): this;
+
+        /**
+         * Specifies that the value must be a multiple of base.
+         */
+        multiple(base: number | Reference): this;
+
+        /**
+         * Requires the number to be negative.
+         */
+        negative(): this;
+
+        /**
+         * Requires the number to be a TCP port, so between 0 and 65535.
+         */
+        port(): this;
+
+        /**
+         * Requires the number to be positive.
+         */
+        positive(): this;
+
+        /**
+         * Specifies the maximum number of decimal places where:
+         * @param limit - the maximum number of decimal places allowed.
+         */
+        precision(limit: number): this;
+
+        /**
+         * Requires the number to be negative or positive.
+         */
+        sign(sign: 'positive' | 'negative'): this;
+
+        /**
+         * Allows the number to be outside of JavaScript's safety range (Number.MIN_SAFE_INTEGER & Number.MAX_SAFE_INTEGER).
+         */
+        unsafe(enabled?: any): this;
+    }
+
+    interface StringSchema extends AnySchema {
+        /**
+         * Requires the string value to only cont
```

Comment:
We should add `flags?: Record<string, Flag>;` here with:
```
    interface Flag {
        setter?: string;
        default?: any
    }
```

(see https://github.com/hapijs/joi/blob/master/lib/schemas.js#L85-L88)